tests/utils.py

```diff
@@ -70,3 +70,17 @@ def get_numpy_array_api_namespace():
         return npx
     else:
         return np
+
+
+def assert_array_equal(
+    actual: np.ndarray,
+    expected: np.ndarray,
+):
+    np.testing.assert_array_equal(
+        actual, expected, strict=np.__version__.startswith("2")
+    )
+    assert isinstance(expected, np.ma.masked_array) ^ (
+        not isinstance(actual, np.ma.masked_array)
+    )
```

Comment:
We want to assert that either both are masked or both are not masked, do we not? I'd find the following easier to read:

```suggestion
    # Assert that either both or neither is masked
    assert isinstance(expected, np.ma.masked_array) == isinstance(actual, np.ma.masked_array)
```